src/resources/addys/claimables/query.ts

```diff
@@ -1,41 +1,43 @@
 import { NativeCurrencyKey } from '@/entities';
-import { QueryConfigWithSelect, QueryFunctionArgs, QueryFunctionResult, createQueryKey } from '@/react-query';
-import { useQuery } from '@tanstack/react-query';
-import { ConsolidatedClaimablesResponse } from './types';
+import { QueryConfigWithSelect, createQueryKey } from '@/react-query';
+import { useQuery, type QueryFunctionContext } from '@tanstack/react-query';
+import { Claimable, ConsolidatedClaimablesResponse } from './types';
 import { logger, RainbowError } from '@/logger';
 import { parseClaimables } from './utils';
 import { useRemoteConfig } from '@/model/remoteConfig';
 import { CLAIMABLES, useExperimentalFlag } from '@/config';
 import { IS_TEST } from '@/env';
 import { useBackendNetworksStore } from '@/state/backendNetworks/backendNetworks';
 import { getAddysHttpClient } from '../client';
+import { Address } from 'viem';
 
 // ///////////////////////////////////////////////
 // Query Types
 
 export type ClaimablesArgs = {
-  address: string;
+  address: Address | string | null;
   currency: NativeCurrencyKey;
+  abortController?: AbortController | null;
 };
 
 // ///////////////////////////////////////////////
 // Query Key
 
-export const claimablesQueryKey = ({ address, currency }: ClaimablesArgs) =>
-  createQueryKey('claimables', { address, currency }, { persisterVersion: 4 });
+export const claimablesQueryKey = ({ address, currency, abortController }: ClaimablesArgs) =>
+  createQueryKey('claimables', { address, currency, abortController }, { persisterVersion: 4 });
 
 type ClaimablesQueryKey = ReturnType<typeof claimablesQueryKey>;
 
-// ///////////////////////////////////////////////
-// Query Function
+const STABLE_CLAIMABLES: ReturnType<typeof parseClaimables<Claimable>> = [];
 
-async function claimablesQueryFunction({ queryKey: [{ address, currency }] }: QueryFunctionArgs<typeof claimablesQueryKey>) {
+export async function getClaimables({ address, currency, abortController }: ClaimablesArgs) {
```

Comment:
Shouldn't this bail if `address` is missing?

Reply:
Unless the `enabled` toggle is rock solid / will never allow this to be called while `address` is unavailable, but still seems safer since it will error.